dnsdist v2.0.5 fails to build with boost v1.91.0

<!-- Hi! Thanks for filing an issue. It will be read with care by human beings. Can we ask you to please fill out this template and not simply demand new features or send in complaints? Thanks! -->
<!-- Also please search the existing issues (both open and closed) to see if your report might be duplicate -->
- [x] This is not a support question, I have read [about opensource](https://www.powerdns.com/opensource.html) and will send support questions to the IRC channel, [GitHub Discussions](https://github.com/PowerDNS/pdns/discussions/) or the mailing list.
- [x] I have read and understood the ['out in the open' support policy](https://blog.powerdns.com/2016/01/18/open-source-support-out-in-the-open/)
- [x] I have read and understood the [PowerDNS AI policy](https://github.com/PowerDNS/pdns/blob/master/AI_POLICY.md)

<!-- Tell us what is issue is about -->
 - Program: dnsdist <!-- delete the ones that do not apply -->
 - Issue type: Bug report

### Short description
dnsdist v2.0.5 fails to build with boost v1.91.0

### Environment
<!-- Tell us about the environment -->
 - Operating system: Arch Linux
 - Software version: dnsdist v2.0.5
 - Software source: compiled myself

### Steps to reproduce
<!-- Tell us step-by-step how the issue can be triggered. Please include your configuration files and any (Lua) scripts that are loaded. -->
1. just trying to build dnsdist v2.0.5 with boost v1.91.0

### Expected behaviour
The build should succeed.

### Actual behaviour
The build process trips here:

```
[  162s] FAILED: [code=1] libdnsdist-common.a.p/dnsdist-lua.cc.o
[  162s] c++ -Ilibdnsdist-common.a.p -I. -I../dnsdist-2.0.5 -Iext/arc4random -I../dnsdist-2.0.5/ext/arc4random -Iext/ipcrypt -I../dnsdist-2.0.5/ext/ipcrypt -Iext/json11 -I../dnsdist-2.0.5/ext/json11 -Iext/lmdb-safe -I../dnsdist-2.0.5/ext/lmdb-safe -I../dnsdist-2.0.5/ext/protozero/include -Iext/yahttp -I../dnsdist-2.0.5/ext/yahttp -Idnsdist-rust-lib -I../dnsdist-2.0.5/dnsdist-rust-lib -Idnsdist-rust-lib/rust -I/usr/include -I/usr/include/p11-kit-1 -I/usr/include/editline -I/usr/include/luajit-2.1 -fdiagnostics-color=always -D_GLIBCXX_ASSERTIONS=1 -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c++17 -DDNSDIST -DHAVE_CONFIG_H -Wshadow -Wmissing-declarations -Wredundant-decls -Wno-ignored-attributes -fvisibility=hidden -fstack-protector --param=ssp-buffer-size=4 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -DBOOST_CONTAINER_USE_STD_EXCEPTIONS -ftrivial-auto-var-init=zero -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -g -ffile-prefix-map=/usr/src/packages/BUILD/dnsdist-hardened/src=/usr/src/debug/dnsdist-hardened -fPIC -pthread -DBOOST_ALL_NO_LIB -MD -MQ libdnsdist-common.a.p/dnsdist-lua.cc.o -MF libdnsdist-common.a.p/dnsdist-lua.cc.o.d -o libdnsdist-common.a.p/dnsdist-lua.cc.o -c ../dnsdist-2.0.5/dnsdist-lua.cc
[  162s] ../dnsdist-2.0.5/dnsdist-lua.cc: In lambda function:
[  163s] ../dnsdist-2.0.5/dnsdist-lua.cc:3087:101: error: converting to ‘boost::optional<std::unordered_map<std::__cxx11::basic_string<char>, boost::variant<bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::hash<std::__cxx11::basic_string<char> >, std::equal_to<std::__cxx11::basic_string<char> >, std::allocator<std::pair<const std::__cxx11::basic_string<char>, boost::variant<bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > > >’ from initializer list would use explicit constructor ‘constexpr boost::optional<T>::optional(U&&) [with U = std::unordered_map<std::__cxx11::basic_string<char>, boost::variant<bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::hash<std::__cxx11::basic_string<char> >, std::equal_to<std::__cxx11::basic_string<char> >, std::allocator<std::pair<const std::__cxx11::basic_string<char>, boost::variant<bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > >&; typename std::enable_if<std::is_constructible<T, U&&>::value, bool>::type <anonymous> = false; typename std::enable_if<(! boost::optional_detail::is_typed_in_place_factory<FT>::value), bool>::type <anonymous> = false; typename std::enable_if<(! boost::optional_detail::is_in_place_factory<FT>::value), bool>::type <anonymous> = false; T = std::unordered_map<std::__cxx11::basic_string<char>, boost::variant<bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::hash<std::__cxx11::basic_string<char> >, std::equal_to<std::__cxx11::basic_string<char> >, std::allocator<std::pair<const std::__cxx11::basic_string<char>, boost::variant<bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > >]’
[  163s]  3087 |         boost::optional<declare_metric_opts_t> vars = {boost::get<declare_metric_opts_t>(opts.get())};
[  163s]       |                                                                                                     ^
[  163s] In file included from /usr/include/boost/optional/optional.hpp:52,
[  163s]                  from /usr/include/boost/optional.hpp:15,
[  163s]                  from ../dnsdist-2.0.5/namespaces.hh:24,
[  163s]                  from ../dnsdist-2.0.5/pdnsexception.hh:26,
[  163s]                  from ../dnsdist-2.0.5/iputils.hh:30,
[  163s]                  from ../dnsdist-2.0.5/bpf-filter.hh:27,
[  163s]                  from ../dnsdist-2.0.5/dnsdist.hh:36,
[  163s]                  from ../dnsdist-2.0.5/dnsdist-lua.cc:35:
[  163s] /usr/include/boost/optional/detail/union_optional.hpp:395:24: note: ‘constexpr boost::optional<T>::optional(U&&) [with U = std::unordered_map<std::__cxx11::basic_string<char>, boost::variant<bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::hash<std::__cxx11::basic_string<char> >, std::equal_to<std::__cxx11::basic_string<char> >, std::allocator<std::pair<const std::__cxx11::basic_string<char>, boost::variant<bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > >&; typename std::enable_if<std::is_constructible<T, U&&>::value, bool>::type <anonymous> = false; typename std::enable_if<(! boost::optional_detail::is_typed_in_place_factory<FT>::value), bool>::type <anonymous> = false; typename std::enable_if<(! boost::optional_detail::is_in_place_factory<FT>::value), bool>::type <anonymous> = false; T = std::unordered_map<std::__cxx11::basic_string<char>, boost::variant<bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::hash<std::__cxx11::basic_string<char> >, std::equal_to<std::__cxx11::basic_string<char> >, std::allocator<std::pair<const std::__cxx11::basic_string<char>, boost::variant<bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > >]’ declared here
[  163s]   395 |     constexpr explicit optional(U&& v)
[  163s]       |                        ^~~~~~~~
[  163s] ../dnsdist-2.0.5/dnsdist-lua.cc: In lambda function:
[  163s] ../dnsdist-2.0.5/dnsdist-lua.cc:3110:94: error: converting to ‘boost::optional<std::unordered_map<std::__cxx11::basic_string<char>, boost::variant<long unsigned int, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > >, std::hash<std::__cxx11::basic_string<char> >, std::equal_to<std::__cxx11::basic_string<char> >, std::allocator<std::pair<const std::__cxx11::basic_string<char>, boost::variant<long unsigned int, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > > > > > >’ from initializer list would use explicit constructor ‘constexpr boost::optional<T>::optional(U&&) [with U = std::unordered_map<std::__cxx11::basic_string<char>, boost::variant<long unsigned int, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > >, std::hash<std::__cxx11::basic_string<char> >, std::equal_to<std::__cxx11::basic_string<char> >, std::allocator<std::pair<const std::__cxx11::basic_string<char>, boost::variant<long unsigned int, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > > > > >&; typename std::enable_if<std::is_constructible<T, U&&>::value, bool>::type <anonymous> = false; typename std::enable_if<(! boost::optional_detail::is_typed_in_place_factory<FT>::value), bool>::type <anonymous> = false; typename std::enable_if<(! boost::optional_detail::is_in_place_factory<FT>::value), bool>::type <anonymous> = false; T = std::unordered_map<std::__cxx11::basic_string<char>, boost::variant<long unsigned int, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > >, std::hash<std::__cxx11::basic_string<char> >, std::equal_to<std::__cxx11::basic_string<char> >, std::allocator<std::pair<const std::__cxx11::basic_string<char>, boost::variant<long unsigned int, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > > > > >]’
[  163s]  3110 |       boost::optional<update_metric_opts_t> vars = {boost::get<update_metric_opts_t>(incOpts)};
[  163s]       |                                                                                              ^
[  163s] /usr/include/boost/optional/detail/union_optional.hpp:395:24: note: ‘constexpr boost::optional<T>::optional(U&&) [with U = std::unordered_map<std::__cxx11::basic_string<char>, boost::variant<long unsigned int, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > >, std::hash<std::__cxx11::basic_string<char> >, std::equal_to<std::__cxx11::basic_string<char> >, std::allocator<std::pair<const std::__cxx11::basic_string<char>, boost::variant<long unsigned int, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > > > > >&; typename std::enable_if<std::is_constructible<T, U&&>::value, bool>::type <anonymous> = false; typename std::enable_if<(! boost::optional_detail::is_typed_in_place_factory<FT>::value), bool>::type <anonymous> = false; typename std::enable_if<(! boost::optional_detail::is_in_place_factory<FT>::value), bool>::type <anonymous> = false; T = std::unordered_map<std::__cxx11::basic_string<char>, boost::variant<long unsigned int, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > >, std::hash<std::__cxx11::basic_string<char> >, std::equal_to<std::__cxx11::basic_string<char> >, std::allocator<std::pair<const std::__cxx11::basic_string<char>, boost::variant<long unsigned int, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > > > > >]’ declared here
[  163s]   395 |     constexpr explicit optional(U&& v)
[  163s]       |                        ^~~~~~~~
[  163s] ../dnsdist-2.0.5/dnsdist-lua.cc: In lambda function:
[  163s] ../dnsdist-2.0.5/dnsdist-lua.cc:3132:94: error: converting to ‘boost::optional<std::unordered_map<std::__cxx11::basic_string<char>, boost::variant<long unsigned int, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > >, std::hash<std::__cxx11::basic_string<char> >, std::equal_to<std::__cxx11::basic_string<char> >, std::allocator<std::pair<const std::__cxx11::basic_string<char>, boost::variant<long unsigned int, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > > > > > >’ from initializer list would use explicit constructor ‘constexpr boost::optional<T>::optional(U&&) [with U = std::unordered_map<std::__cxx11::basic_string<char>, boost::variant<long unsigned int, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > >, std::hash<std::__cxx11::basic_string<char> >, std::equal_to<std::__cxx11::basic_string<char> >, std::allocator<std::pair<const std::__cxx11::basic_string<char>, boost::variant<long unsigned int, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > > > > >&; typename std::enable_if<std::is_constructible<T, U&&>::value, bool>::type <anonymous> = false; typename std::enable_if<(! boost::optional_detail::is_typed_in_place_factory<FT>::value), bool>::type <anonymous> = false; typename std::enable_if<(! boost::optional_detail::is_in_place_factory<FT>::value), bool>::type <anonymous> = false; T = std::unordered_map<std::__cxx11::basic_string<char>, boost::variant<long unsigned int, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > >, std::hash<std::__cxx11::basic_string<char> >, std::equal_to<std::__cxx11::basic_string<char> >, std::allocator<std::pair<const std::__cxx11::basic_string<char>, boost::variant<long unsigned int, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > > > > >]’
[  163s]  3132 |       boost::optional<update_metric_opts_t> vars = {boost::get<update_metric_opts_t>(decOpts)};
[  163s]       |                                                                                              ^
[  163s] /usr/include/boost/optional/detail/union_optional.hpp:395:24: note: ‘constexpr boost::optional<T>::optional(U&&) [with U = std::unordered_map<std::__cxx11::basic_string<char>, boost::variant<long unsigned int, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > >, std::hash<std::__cxx11::basic_string<char> >, std::equal_to<std::__cxx11::basic_string<char> >, std::allocator<std::pair<const std::__cxx11::basic_string<char>, boost::variant<long unsigned int, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > > > > >&; typename std::enable_if<std::is_constructible<T, U&&>::value, bool>::type <anonymous> = false; typename std::enable_if<(! boost::optional_detail::is_typed_in_place_factory<FT>::value), bool>::type <anonymous> = false; typename std::enable_if<(! boost::optional_detail::is_in_place_factory<FT>::value), bool>::type <anonymous> = false; T = std::unordered_map<std::__cxx11::basic_string<char>, boost::variant<long unsigned int, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > >, std::hash<std::__cxx11::basic_string<char> >, std::equal_to<std::__cxx11::basic_string<char> >, std::allocator<std::pair<const std::__cxx11::basic_string<char>, boost::variant<long unsigned int, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > > > > >]’ declared here
[  163s]   395 |     constexpr explicit optional(U&& v)
[  163s]       |                        ^~~~~~~~
```

### Other information
Asked about this in IRC before posting a report, nothing more ATM.

--- a/pdns/dnsdistdist/dnsdist-lua.cc
+++ b/pdns/dnsdistdist/dnsdist-lua.cc
@@ -3214,7 +3214,7 @@ static void setupLuaConfig(LuaContext& luaCtx, bool client, bool configCheck)
         customName = std::optional(*optCustomName);
       }
       if (!customName) {
-        std::optional<declare_metric_opts_t> vars = {boost::get<declare_metric_opts_t>(opts.value())};
+        std::optional<declare_metric_opts_t> vars{boost::get<declare_metric_opts_t>(opts.value())};
         getOptionalValue<std::string>(vars, "customName", customName);
         getOptionalValue<bool>(vars, "withLabels", withLabels);
         checkAllParametersConsumed("declareMetric", vars);
@@ -3238,7 +3238,7 @@ static void setupLuaConfig(LuaContext& luaCtx, bool client, bool configCheck)
       step = *custom_step;
     }
     else {
-      std::optional<update_metric_opts_t> vars = {boost::get<update_metric_opts_t>(incOpts)};
+      std::optional<update_metric_opts_t> vars{boost::get<update_metric_opts_t>(incOpts)};
       getOptionalValue<uint64_t>(vars, "step", step);
       getOptionalValue<LuaAssociativeTable<std::string>>(vars, "labels", labels);
       checkAllParametersConsumed("incMetric", vars);
@@ -3261,7 +3261,7 @@ static void setupLuaConfig(LuaContext& luaCtx, bool client, bool configCheck)
       step = *custom_step;
     }
     else {
-      std::optional<update_metric_opts_t> vars = {boost::get<update_metric_opts_t>(decOpts)};
+      std::optional<update_metric_opts_t> vars{boost::get<update_metric_opts_t>(decOpts)};
       getOptionalValue<uint64_t>(vars, "step", step);
       getOptionalValue<LuaAssociativeTable<std::string>>(vars, "labels", labels);
       checkAllParametersConsumed("decMetric", vars);

